.github/workflows/docker-build-test-upload-all.yml

```diff
@@ -0,0 +1,244 @@
+name: Build, test, and push Docker Images
```

Comment:
I would have preferred having this refactoring in a separate PR first, but I'm ok keeping it here.
I have another concern though  - can we make left side of the run better?
With this PR, I need to click all build to find `datascience` build, for example.

<img width="318" alt="Screenshot 2024-02-08 at 13 25 08" src="https://github.com/jupyter/docker-stacks/assets/12270691/8d0a233d-1a4e-4482-85c6-249b8052fc0c">

Reply:
I also think you can still create a separate PR for this change (having `docker-build-test-upload-all.yml` file), then this PR will look much more readable.

Reply:
I reverted the refactor.